nuget/lib/dependabot/nuget/nuget_config_credential_helpers.rb

```diff
@@ -48,6 +50,36 @@ def self.add_credentials_to_nuget_config(credentials)
         File.write(user_nuget_config_path, nuget_config)
       end
 
+      sig { params(credentials: T::Array[Dependabot::Credential]).returns(T::Hash[String, String]) }
+      def self.get_credentials_to_credential_helper_env(credentials)
+        env = {}
+
+        nuget_credentials = credentials.select { |cred| cred["type"] == "nuget_feed" }
+
+        if nuget_credentials.any?
+          endpoint_credentials = []
+
+          nuget_credentials.each do |c|
+            next unless c["token"]
+
+            exploded_token = T.must(c["token"]).split(":", 2)
+
+            next unless exploded_token.length == 2
+
+            username = exploded_token[0]
+            password = exploded_token[1]
+
+            endpoint_credentials << <<~NUGET_ENDPOINT_CREDENTIAL
+              {"endpoint":"#{c['url']}", "username":"#{username}", "password":"#{password}"}
```

Comment:
Here and a few lines below, use the `.to_json` method so we don't have to worry about any escaping.  I'm not 100% certain on the syntax, but it's something like this:

``` ruby
endpoint_credentials << {
  "endpoint" => c['url'],
  "username" => username,
  "password" => password,
}
```

Then later on:

``` ruby
env["VSS_NUGET_EXTERNAL_FEED_ENDPOINTS"] = {
 "endpointCredentials" => endpoint_credentials
}.to_json
```